### PR TITLE
Anthropic server tool result returned to attributes

### PR DESCRIPTION
**Pull request by ejstuart**
Follow-up to https://github.com/langchain4j/langchain4j/pull/4211

## Change
Include anthropic server tools results in AnthropicContent and then map them into the attributes of AiMessage when enabled in ChatModel.

## General checklist

- [X] There are no breaking changes (API, behaviour)
- [X] I have added unit and/or integration tests for my change
- [X] The tests cover both positive and negative cases (existing tests cover negative cases)
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [X] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)

### CODE REVIEW

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicChatModel.java [hunk 1]
```diff
@@ -70,6 +70,7 @@ public class AnthropicChatModel implements ChatModel {
     private final String thinkingType;
     private final Integer thinkingBudgetTokens;
     private final boolean returnThinking;
+    private final boolean returnServerToolResults;
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicChatModel.java]: please put it after `serverTools`, it is currently in the middle of "thinking" fields. Same for all other places in this PR

File: langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicChatModelIT.java [hunk 2]
```diff
@@ -764,11 +770,47 @@ void should_send_strict_true_in_tools_definition() {
...
         assertThat(call.name()).isEqualTo("get_weather");
 
-        assertThat(call.arguments())
-                .contains("\"location\"")
-                .doesNotContain("\"unit\"");
+        assertThat(call.arguments()).contains("\"location\"").doesNotContain("\"unit\"");
     }
 
+    @Test
+    void should_return_server_tool_results_in_attributes_when_enabled() {
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicChatModelIT.java:777]: please add a similar test to the `AnthropicStreamingChatModelIT`

**Review by dliubarskyi — COMMENTED** [review]
@ejstuart well done, thank you!

Please document it under https://docs.langchain4j.dev/integrations/language-models/anthropic#server-tools and https://docs.langchain4j.dev/integrations/language-models/anthropic#customizing-anthropicchatmodel sections

**Comment by ejstuart** [comment]
@dliubarskyi Done! Let me know if there's anything else

**Review by dliubarskyi — APPROVED** [review]
@ejstuart thank you!
